exchanges/huobihadax/huobihadax_types.go

```diff
@@ -321,3 +325,195 @@ type WsTrade struct {
+	SignatureVersion string        `json:"SignatureVersion"`
+	Timestamp        string        `json:"Timestamp"`
+	Signature        string        `json:"Signature"`
+	Topic            string        `json:"topic"`
+	States           string        `json:"states"`
+	AccountID        int64         `json:"account-id"`
+	Symbol           currency.Pair `json:"symbol"`
+}
+
+// WsAuthenticatedDataResponse response from authenticated connection
+type WsAuthenticatedDataResponse struct {
+	Op           string `json:"op,omitempty"`
+	Ts           int64  `json:"ts,omitempty"`
+	Topic        string `json:"topic,omitempty"`
+	ErrorCode    int64  `json:"err-code,omitempty"`
+	ErrorMessage string `json:"err-msg,omitempty"`
+	Ping         int64  `json:"ping,omitempty"`
+	CID          string `json:"cid,omitempty"`
+}
+
+// WsAuthenticatedAccountsResponse response from Accounts authenticated subscription
+type WsAuthenticatedAccountsResponse struct {
+	WsAuthenticatedDataResponse
+	Data WsAuthenticatedAccountsResponseData `json:"data"`
+}
+
+// WsAuthenticatedAccountsResponseData account data
+type WsAuthenticatedAccountsResponseData struct {
+	Event string                                    `json:"event"`
+	List  []WsAuthenticatedAccountsResponseDataList `json:"list"`
+}
+
+// WsAuthenticatedAccountsResponseDataList detailed account data
+type WsAuthenticatedAccountsResponseDataList struct {
+	AccountID int64  `json:"account-id"`
+	Currency  string `json:"currency"`
+	Type      string `json:"type"`
+	Balance   string `json:"balance"`
+}
+
+// WsAuthenticatedOrdersUpdateResponse response from OrdersUpdate authenticated subscription
+type WsAuthenticatedOrdersUpdateResponse struct {
+	WsAuthenticatedDataResponse
+	Data WsAuthenticatedOrdersUpdateResponseData `json:"data"`
+}
+
+// WsAuthenticatedOrdersUpdateResponseData order  updatedata
+type WsAuthenticatedOrdersUpdateResponseData struct {
+	UnfilledAmount   string        `json:"unfilled-amount"`
+	FilledAmount     string        `json:"filled-amount"`
+	Price            string        `json:"price"`
+	OrderID          int64         `json:"order-id"`
+	Symbol           currency.Pair `json:"symbol"`
+	MatchID          int64         `json:"match-id"`
+	FilledCashAmount string        `json:"filled-cash-amount"`
+	Role             string        `json:"role"`
+	OrderState       string        `json:"order-state"`
+}
+
+// WsAuthenticatedOrdersResponse response from Orders authenticated subscription
+type WsAuthenticatedOrdersResponse struct {
+	WsAuthenticatedDataResponse
+	Data []WsAuthenticatedOrdersResponseData `json:"data"`
+}
+
+// WsAuthenticatedOrdersResponseData order data
+type WsAuthenticatedOrdersResponseData struct {
+	SeqID            int64         `json:"seq-id"`
+	OrderID          int64         `json:"order-id"`
+	Symbol           currency.Pair `json:"symbol"`
+	AccountID        int64         `json:"account-id"`
+	OrderAmount      string        `json:"order-amount"`
```

Comment:
OrderAmont, OrderPrice, FilledAmount, UnfilledAmount, FilledCashAmount and FilledFees can be float64s